src/devices/RGBDSensorClient/RGBDSensorClient_StreamingMsgParser.cpp

```diff
@@ -9,67 +9,97 @@
 
 #include "RGBDSensorClient_StreamingMsgParser.h"
 
+#include <yarp/os/Time.h>
+
 using namespace yarp::dev;
 
 // Callback reader for rgb
-RgbImageReader_Impl::RgbImageReader_Impl() = default;
-RgbImageReader_Impl::~RgbImageReader_Impl() = default;
-
-void RgbImageReader_Impl::onRead(yarp::sig::FlexImage& datum)
+void RgbImageBufferedPort::onRead(yarp::sig::FlexImage& datum)
 {
-    last_rgb = datum;
+    local_arrival_time = yarp::os::Time::now();
+    std::lock_guard<std::mutex> lock(mutex);
```

Comment:
The lock is acquired after modifying `local_arrival_time`. Is this intentional? Note that in `getImage` this variable is accessed within the critical section, hence my doubt.